meta-balena-common/recipes-containers/balena-supervisor/balena-supervisor/update-balena-supervisor

```diff
@@ -99,30 +116,39 @@ fi
 echo "Getting image name and version..."
 if [ -n "$API_ENDPOINT" ] && [ -n "${UUID}" ] && [ -n "$_device_api_key" ]; then
     _target_state=$(os_helpers_fetch_target_state "${UUID}" "${API_ENDPOINT}" "${_device_api_key}")
-    read -r version image_name <<< "${_target_state}"
-    if [ -z "$version" ] || [ -z "$image_name" ]; then
-        echo "ERROR: Unexpected API data."
-        error_handler "unexpected API data"
+    if [ -n "${_target_state}" ]; then
+        read -r version image_name <<< "${_target_state}"
+        if [ -z "$version" ] || [ -z "$image_name" ]; then
+            echo "ERROR: Unexpected API data."
+            error_handler "unexpected API data"
+        fi
     fi
 fi
 
 # shellcheck disable=SC1091
 . /etc/balena-supervisor/supervisor.conf
 
 # If no API version is set, use preloaded values
-if [ "$version" = "null" ] || [ "$image_name" = "null" ]; then
-    echo "No supervisor configuration found from API. Using preloaded values."
-    if [ -z "$SUPERVISOR_VERSION" ]; then
-        echo "ERROR: No preloaded version found."
-        error_handler "no preloaded version found"
-    fi
-    if [ -z "$SUPERVISOR_IMAGE" ]; then
-        echo "ERROR: No preloaded image found."
-        error_handler "no preloaded image found."
+if [ -z "$version" ] || [ -z "$image_name" ] || [ "$version" = "null" ] || [ "$image_name" = "null" ]; then
+    echo "No supervisor configuration found from API."
+    if [ -n "${TARGET_SUPERVISOR_IMAGE}" ]; then
+        echo "Using command line image argument $TARGET_SUPERVISOR_IMAGE."
+        image_name="${TARGET_SUPERVISOR_IMAGE%:*}"
+        version=$(basename "${image_name}")
```

Comment:
@klutchell  The only case when we can obtain a version is for manages devices via an API call once the v3 target state reports the supervisor version (the current `supervisor_version` endpoint reports `latest`).
In all other cases we can either default to `latest` or use something from the image path. For the new syntax, this will be the commit|semver, for an image path it's the digest you mention which is at least better than `latest` to identify it.
Any other ideas welcomed though.

Reply:
This is fine with me, it just looks weird when using an image path with the digest but it's all the info we have at that point so it's the best we can do. Agreed that it's better than `latest`!